generate/convert.go

```diff
@@ -253,6 +261,18 @@ func (g *generator) convertType(
 	return goTyp, err
 }
 
+// getWeakReference decides if a field should be of pointer type and have the omitempty flag set.
+func (g *generator) getWeakReference(
+	options *genqlientDirective,
+	isStructField bool,
+	def *ast.Definition,
+) bool {
+	return options.Pointer == nil &&
+		isStructField &&
+		g.Config.WeakReferences &&
+		len(def.Fields) > 0
```

Comment:
Oh so actually a question here is whether `use_weak_references` should apply to interface-typed fields.  Unless you see a strong reason, my feeling is not -- pointer-to-interface types are super awkward in Go.  In that case this should be `def.Kind == ast.Object || def.Kind == ast.InputObject` (or something like that).)

Reply:
Great catch, I hadn't considered the interface-typed fields.  I don't see why we shouldn't use pointers for those, that way the rule doesn't have any gotchas

Reply:
They just tend to be awkward in that you have to explicitly dereference to call the interface's methods (i.e. `(*resp.F).M()`) and confusing in that the value inside the interface is often itself a pointer.  I think it's clean to explain either way -- if anything since the option is "struct_references" that would suggest it's just structs.  We could rename it and do interfaces but I feel that's only going to be annoying.

Another question that we had discussed in the issue is whether it applies to output objects.  I think "what is easiest to explain" is also a good criterion there, but it could go either way (if inputs, we'd call it "input_references" and it makes the interfaces question moot since GraphQL doesn't have interfaces in inputs).